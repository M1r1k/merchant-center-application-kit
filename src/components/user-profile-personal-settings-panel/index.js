export { default } from './user-profile-personal-settings-panel';
