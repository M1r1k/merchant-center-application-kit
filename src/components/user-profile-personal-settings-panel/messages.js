import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'UserProfile.PersonalSettingsPanel.title',
    description: 'Title of the User Profile personal settings panel',
    defaultMessage: 'Personal Settings',
  },
  language: {
    id: 'UserProfile.PersonalSettingsPanel.language',
    description: 'Language label',
    defaultMessage: 'Language',
  },
  timeZone: {
    id: 'UserProfile.PersonalSettingsPanel.timeZone',
    description: 'TimeZone label',
    defaultMessage: 'Time Zone',
  },
});
