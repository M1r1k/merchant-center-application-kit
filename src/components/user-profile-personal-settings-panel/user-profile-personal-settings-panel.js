import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import { connect } from 'react-redux';
import { compose } from 'recompose';
import { defaultMemoize } from 'reselect';
import Select from 'react-select';
import { Field, formValueSelector } from 'redux-form';
import {
  withTimeZones,
  timeZonesShape,
} from '@commercetools-local/l10n/time-zone-information';
import wrapInputForReduxForm from '@commercetools-local/core/components/input-wrapper';
import CollapsiblePanel from '@commercetools-local/core/components/toggles/collapsible-panel';
import FormBox from '@commercetools-local/core/components/form-box';
import LabelField from '@commercetools-local/core/components/fields/label-field';
import { USER_PROFILE_FORM_NAME } from '../../constants';
import messages from './messages';

const RFSelect = wrapInputForReduxForm(Select);

export const timeZonesToOptions = defaultMemoize(timeZones =>
  Object.entries(timeZones).map(([code, value]) => ({
    value: code,
    // E.g. `Europe/Berlin - CEST (+02:00)`
    label: `${code} - ${value.abbr} (${value.offset})`,
  }))
);

export const UserProfilePersonalSettingsPanel = props => (
  <CollapsiblePanel label={props.intl.formatMessage(messages.title)}>
    <FormBox>
      <LabelField
        title={props.intl.formatMessage(messages.language)}
        isRequired={true}
      />
      <Field
        name="language"
        component={RFSelect}
        options={[{ value: 'en', label: 'EN' }, { value: 'de', label: 'DE' }]}
        clearable={false}
        searchable={false}
        parse={
          /* transform react select option to form value */
          ({ value }) => value
        }
      />
    </FormBox>
    <FormBox>
      <LabelField title={props.intl.formatMessage(messages.timeZone)} />
      <Field
        name="timeZone"
        component={RFSelect}
        options={timeZonesToOptions(props.timeZones)}
        clearable={true}
        searchable={true}
        parse={
          /* transform react select option to form value */
          option => (option ? option.value : null)
        }
      />
    </FormBox>
  </CollapsiblePanel>
);
UserProfilePersonalSettingsPanel.displayName =
  'UserProfilePersonalSettingsPanel';
UserProfilePersonalSettingsPanel.propTypes = {
  hasSubmitFailed: PropTypes.bool,
  // HoC
  timeZones: timeZonesShape,
  intl: intlShape.isRequired,
};

const valueSelector = formValueSelector(USER_PROFILE_FORM_NAME);

export default compose(
  connect(state => ({ locale: valueSelector(state, 'language') })),
  withTimeZones(ownProps => ownProps.locale),
  injectIntl
)(UserProfilePersonalSettingsPanel);
