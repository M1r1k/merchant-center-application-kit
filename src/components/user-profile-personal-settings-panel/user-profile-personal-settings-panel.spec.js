import React from 'react';
import { shallow } from 'enzyme';
import { intlMock } from '@commercetools-local/test-utils';
import { UserProfilePersonalSettingsPanel } from './user-profile-personal-settings-panel';

const createTestProps = props => ({
  hasSubmitFailed: false,
  timeZones: {
    'Europe/Berlin': { name: 'Europe/Berlin', abbr: 'CEST', offset: '+02:00' },
  },
  intl: intlMock,
  ...props,
});

describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<UserProfilePersonalSettingsPanel {...props} />);
  });
  it('should ensure layout structure', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render <Field> for language', () => {
    expect(wrapper).toRender({ name: 'language' });
  });
  it('should render <Field> for timeZone', () => {
    expect(wrapper).toRender({ name: 'timeZone' });
  });
});
