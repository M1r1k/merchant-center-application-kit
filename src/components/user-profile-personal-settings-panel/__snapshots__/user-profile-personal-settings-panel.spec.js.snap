// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should ensure layout structure 1`] = `
<CollapsiblePanel
  label="UserProfile.PersonalSettingsPanel.title"
  tone="primary"
>
  <FormBox
    hasWarning={false}
  >
    <LabelField
      isRequired={true}
      title="UserProfile.PersonalSettingsPanel.language"
    />
    <Field
      clearable={false}
      component={[Function]}
      name="language"
      options={
        Array [
          Object {
            "label": "EN",
            "value": "en",
          },
          Object {
            "label": "DE",
            "value": "de",
          },
        ]
      }
      parse={[Function]}
      searchable={false}
    />
  </FormBox>
  <FormBox
    hasWarning={false}
  >
    <LabelField
      isRequired={false}
      title="UserProfile.PersonalSettingsPanel.timeZone"
    />
    <Field
      clearable={true}
      component={[Function]}
      name="timeZone"
      options={
        Array [
          Object {
            "label": "Europe/Berlin - CEST (+02:00)",
            "value": "Europe/Berlin",
          },
        ]
      }
      parse={[Function]}
      searchable={true}
    />
  </FormBox>
</CollapsiblePanel>
`;
